### PR TITLE
SIRo Sandbox tool MVP

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

See examples/siro_sandbox/README.md for details.

siro_sandbox_screenshot

## How Has This Been Tested

Local testing. 

## Types of changes

This PR merges into `SIRo`, not `main`. This would be a Development-type change if/when it merges to `main`.

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

**Comment by rpartsey** [comment]
Videos of running HITL eval with a user-controlled humanoid and policy-driven Fetch

**Mac M1**
Sandbox Tool (video is too large to be uploaded in GitHub comment):

https://drive.google.com/file/d/1vNSv3Tx9PsKbYQHZk2aFJT9KZHDeDJ_S/view?usp=share_link

Sandbox Tool + batch renderer fail on my Mac M1 with the following error:
Screenshot 2023-04-06 at 17 58 23

**ubuntu22.04**
Sandbox Tool:

[hitl_ubuntu22.04.webm](https://user-images.githubusercontent.com/37709869/230472679-2e0afb7b-62b8-452f-9c5d-2e27efb18c09.webm)

Sandbox Tool + batch renderer (**note**, that the control is not working properly; blue trajectory that humanoid is expected to follow doesn't "repeat" mouse movement):

[hitl_batch_renderer_ubuntu22.04.webm](https://user-images.githubusercontent.com/37709869/230472593-7d4499c2-45ee-4333-bfb3-7817903bfb1c.webm)

**Comment by rpartsey** [comment]
@eundersander  In general everything looks great to me. Very enjoyable to control the humanoid and I like how everything looks visually.

The only 2 things I'm worried about are:

1. running Sandbox Tool + batch renderer fail on my Mac M1 machine with the error I've attached in the above comment (but I don't know if it is specific only to my machine)
2. in Sandbox Tool + batch renderer (on ubuntu) blue trajectory doesn't follow the movement of the mouse

File: examples/siro_sandbox/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Known Issues
+* The policy-driven agent doesn't seem to be working in terms of producing interesting actions. As a placeholder, I've injected random-base-movement behavior in `BaselinesController.act`; see comment "temp do random base actions".
+* One-time visual flicker shortly after app startup on Mac
+
+## Running HITL eval with a user-controlled humanoid and policy-driven Fetch
+
+1. Make sure you've followed the [SIRo install instructions](../../SIRO_README.md#installation).
+2. Run this command:
+```
+HABITAT_SIM_LOG=quiet python examples/siro_sandbox/sandbox_app.py --disable-inverse-kinematics --humanoid-user-agent --cfg habitat-lab/habitat/config/benchmark/rearrange/rearrange_easy_human_and_fetch.yaml --never-end habitat.dataset.split=minival
```
Comment (rpartsey) [inline examples/siro_sandbox/README.md:17]: `habitat-lab/habitat/config/` config path prefix can be removed. Hydra searches config files relative to `habitat-lab/habitat/config` directory.

File: examples/siro_sandbox/controllers.py [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+        self.controllers = []
+        self.n_robots = self.n_robots
+        self.all_hxs = [None for _ in range(self.n_robots)]
+        self.active_controllers = [0, 1]
+        self.controllers = [
+            gui_controller,
+            BaselinesController(
+                1,
+                is_multi_agent,
+                "habitat-baselines/habitat_baselines/config/rearrange/rl_hierarchical.yaml",
```
Comment (rpartsey) [inline examples/siro_sandbox/controllers.py:516]: `habitat-baselines/habitat_baselines/config/` config path prefix can be removed.

File: examples/siro_sandbox/sandbox_app.py [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+    PddlApplyActionConfig,
+)
+from habitat.gui.gui_application import GuiAppDriver, GuiApplication
+from habitat.gui.gui_input import GuiInput
+from habitat.gui.replay_gui_app_renderer import ReplayGuiAppRenderer
+
+# Please reach out to the paper authors to obtain this file
+DEFAULT_POSE_PATH = "data/humanoids/humanoid_data/walking_motion_processed.pkl"
+
+DEFAULT_CFG = "habitat-lab/habitat/config/benchmark/rearrange/rearrange_easy_human_and_fetch.yaml"
```
Comment (rpartsey) [inline examples/siro_sandbox/sandbox_app.py:42]: `habitat-lab/habitat/config/` config path prefix can be removed.

**Comment by eundersander** [comment]
@rpartsey Thanks for high-level comments above. Let me address a few here:

>Sandbox Tool + batch renderer fail on my Mac M1 machine
This is expected and a fundamental limitation of the batch renderer. I will add a note in the README.

>in Sandbox Tool + batch renderer (on ubuntu) blue trajectory doesn't follow the movement of the mouse
The `--use-batch-renderer` flag is experimental and really just for Mikael and me for debugging, so I'm not gonna worry about this right now.

>it would be nice to also enable scene rotation in addition to zoom the camera in/out
Absolutely. Let's aim to add UX improvements like this in follow-up PRs. For this PR review, let's limit the scope to code organization and bugs that affect most users.

**Review by rpartsey — APPROVED** [review]

**Pull request merged**